3135/2/user_info_class_set.rb

```diff
@@ -0,0 +1,160 @@
+# User class provides access to user info and manages immediate properties
+#:reek:TooManyInstanceVariables
+class User
+  attr_reader :id, :camp_num, :location, :photo_uri, :action, :request, :save
+
+  def initialize(tg_id)
+    @id = tg_id
+    @camp_num = R.get("tgid_#{tg_id}_camp_num")
+    @location = R.get("tgid_#{tg_id}_location")
+    @photo_uri = R.get("tgid_#{tg_id}_photo_uri")
+
+    @action = ActionStatus.new(tg_id)
+    @request = RequestStatus.new(tg_id)
+    @save = DataSaver.new(tg_id)
+  end
+
+  # -residency
+  def resident?
+    R.get("tgid_#{id}_rank") == 'resident'
+  end
+
+  def give_residency
+    R.set("tgid_#{id}_rank", 'resident')
+  end
+
+  # -presence
+  def presence_init
+    R.set("tgid_#{id}_presence", 'offsite')
+  end
+
+  def present?
+    R.get("tgid_#{id}_presence") == 'onsite'
+  end
+
+  def presence_switch
+    if present?
+      R.set("tgid_#{id}_presence", 'offsite')
+    else
+      R.set("tgid_#{id}_presence", 'onsite')
+    end
+  end
+
+  # -status flush
+  def status_flush
+    @action.flush
+    @request.flush
+  end
+end
+
+# ActionStatus class manages currently ongoing action and stores their statuses to redis
+class ActionStatus
```

Comment:
Ruby style guide says 'One Class per File'
However, if you are grouping classes with related functionality into a single file then you should also consider making them part of the same module for a namespace.
Related to all other files